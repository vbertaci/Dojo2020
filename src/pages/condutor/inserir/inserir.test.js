import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Inserir from './inserir';

Enzyme.configure({ adapter: new Adapter() });

const wrapper = mount(<Inserir />)

describe('Teste da interface "Detalhes"', () => {
    it('Deve renderizar o componente e comparar com um snapshot', () => {
        const wrapperSnapshot = shallow(<Inserir />)
        expect(wrapperSnapshot.debug()).toMatchSnapshot();
    })

    it('Valida o campo nome', () => {

        const nome = 'Coding Jojo';

        const inputNome = wrapper.find('#nome');
        inputNome.instance().value = nome;
        inputNome.simulate('change');

        expect(wrapper.state().usuario.nome).toEqual(nome);
    })

    it('Valida o campo cpf', () => {

        const cpf = '___.___.___-__';

        const inputCpf = wrapper.find('input#cpf').at(0);
        inputCpf.instance().value = cpf;
        inputCpf.simulate('change');

        expect(wrapper.state().usuario.cpf).toEqual(cpf);
    })

    it('Valida o campo registro', () => {

        const registro = '___________';

        const inputRegistro = wrapper.find('#registro');
        inputRegistro.instance().value = registro;
        inputRegistro.simulate('change');

        expect(wrapper.state().usuario.registro).toEqual(registro);
    })

    it('Valida o campo validade', () => {

        const validade = '2020-10-10';

        const inputValidade = wrapper.find('#validade');
        inputValidade.instance().value = validade;
        inputValidade.simulate('change');

        expect(wrapper.state().usuario.validade).toEqual(validade);
    })

    it('Valida o campo email', () => {

        const email = 'email@dominio.com';

        const inputEmail = wrapper.find('#email');
        inputEmail.instance().value = email;
        inputEmail.simulate('change');

        expect(wrapper.state().usuario.email).toEqual(email);
    })

})
