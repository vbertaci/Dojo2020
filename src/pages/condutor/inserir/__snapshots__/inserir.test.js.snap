// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste da interface "Detalhes" Deve renderizar o componente e comparar com um snapshot 1`] = `
"<section className=\\"new-user\\">
  <div className=\\"container\\">
    <form onSubmit={[Function]}>
      <fieldset>
        <div className=\\"d-flex align-items-center mb-4\\">
          <h1>
            Cadastro de condutor
          </h1>
          <a className=\\"btn btn-light ml-auto back-link\\" href=\\"/\\">
            Voltar
          </a>
        </div>
        <div className=\\"form-group\\">
          <label htmlFor=\\"nome\\">
            Nome
          </label>
          <input className=\\"form-control\\" type=\\"text\\" id=\\"nome\\" name=\\"nome\\" value=\\"\\" onChange={[Function]} required={true} />
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-6\\">
            <div className=\\"form-group\\">
              <label htmlFor=\\"cpf\\">
                CPF
              </label>
              <InputMask mask=\\"999.999.999-99\\" className=\\"form-control\\" type=\\"text\\" id=\\"cpf\\" name=\\"cpf\\" value=\\"\\" onChange={[Function]} required={true} alwaysShowMask={false} maskPlaceholder=\\"_\\" />
            </div>
          </div>
          <div className=\\"col-sm-6\\">
            <div className=\\"form-group\\">
              <label htmlFor=\\"registro\\">
                Registro
              </label>
              <InputMask mask=\\"99999999999\\" className=\\"form-control\\" type=\\"text\\" id=\\"registro\\" name=\\"registro\\" value=\\"\\" onChange={[Function]} required={true} alwaysShowMask={false} maskPlaceholder=\\"_\\" />
            </div>
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3\\">
            <div className=\\"form-group\\">
              <label htmlFor=\\"categoria\\">
                Categoria
              </label>
              <select className=\\"form-control\\" id=\\"categoria\\" name=\\"categoria\\" onChange={[Function]} required={true}>
                <option>
                  Selecione
                </option>
                <option value=\\"A\\">
                  A
                </option>
                <option value=\\"AB\\">
                  AB
                </option>
                <option value=\\"AC\\">
                  AC
                </option>
                <option value=\\"AD\\">
                  AD
                </option>
                <option value=\\"AE\\">
                  AE
                </option>
                <option value=\\"ACC\\">
                  ACC
                </option>
                <option value=\\"B\\">
                  B
                </option>
                <option value=\\"C\\">
                  C
                </option>
                <option value=\\"D\\">
                  D
                </option>
                <option value=\\"E\\">
                  E
                </option>
              </select>
            </div>
          </div>
          <div className=\\"col-sm-3\\">
            <div className=\\"form-group\\">
              <label htmlFor=\\"validade\\">
                Validade
              </label>
              <input className=\\"form-control\\" type=\\"date\\" id=\\"validade\\" name=\\"validade\\" value=\\"\\" onChange={[Function]} required={true} max=\\"2030-12-31\\" />
            </div>
          </div>
          <div className=\\"col-sm-6\\">
            <div className=\\"form-group\\">
              <label htmlFor=\\"email\\">
                Email
              </label>
              <input className=\\"form-control\\" type=\\"email\\" id=\\"email\\" name=\\"email\\" value=\\"\\" onChange={[Function]} required={true} />
            </div>
          </div>
        </div>
        <div className=\\"form-group mt-3\\">
          <button id=\\"btn-inserir\\" className=\\"btn btn-primary btn-lg btn-block\\">
            Inserir
          </button>
        </div>
      </fieldset>
    </form>
  </div>
</section>"
`;
